openbas-api/src/main/java/io/openbas/executors/crowdstrike/service/CrowdStrikeExecutorService.java

```diff
@@ -91,16 +102,50 @@ public CrowdStrikeExecutorService(
   @Override
   public void run() {
     log.info("Running CrowdStrike executor endpoints gathering...");
-    List<CrowdStrikeDevice> devices =
-        this.client.devices().getResources().stream()
-            .filter(device -> device.getHostname() != null)
-            .toList();
-    List<AgentRegisterInput> endpointRegisterList = toAgentEndpoint(devices);
-    log.info(
-        "CrowdStrike executor provisioning based on " + endpointRegisterList.size() + " assets");
-
-    for (AgentRegisterInput input : endpointRegisterList) {
-      endpointService.registerAgentEndpoint(input);
+    List<String> hostGroups = Stream.of(this.config.getHostGroup().split(",")).distinct().toList();
+    ResourcesGroups crowdStrikeResourceGroup;
+    CrowdStrikeHostGroup crowdStrikeHostGroup;
+    for (String hostGroup : hostGroups) {
+      crowdStrikeResourceGroup = this.client.hostGroup(hostGroup);
+      if (crowdStrikeResourceGroup.getErrors() != null
+          && !crowdStrikeResourceGroup.getErrors().isEmpty()) {
+        CrowdstrikeError e = crowdStrikeResourceGroup.getErrors().getFirst();
+        log.log(
+            Level.SEVERE,
+            "Error occurred while getting Crowdstrike hostGroup API request for id "
+                + hostGroup
+                + ". Code: "
+                + e.getCode()
+                + ", message: "
+                + e.getMessage());
+        continue;
+      }
+      List<CrowdStrikeDevice> devices = this.client.devices(hostGroup);
```

Comment:
Again, nitpicking here but could you add error handling here like you did above ?